app/controllers/tasks_controller.rb

```diff
@@ -1,18 +1,23 @@
 class TasksController < ApplicationController
 
   def index
+    Task.populate_from_airtable(current_user.email)
     @filter = params[:category]
-    @all_categories = %w[Inventory Register Engineering]
+    @all_categories = Setting.last.categories.split(/\s*,\s*/)
     @tasks = if @filter.blank?
-               Task.all
+               Task.where('assigneduserstring LIKE ?', '%' + current_user.email + '%').or(Task.where(priority: 3)).or(Task.where(priority: 1)).all
```

Comment:
this query needs to be cleaned up. there's a duplicate of `where(category: @filter)` in line 10-11.  the common part about assigneduserstring should be factored outside the conditional, since you can build up a query over several steps.  in general, this logic is kind of illegible and needs to be broken down into easier-to-follow steps.

Reply:
resolved, logic should be much clearer now and repetitions were factored out